### PR TITLE
Memoize `Site#site_data`

### PR DESCRIPTION
**Pull request by ashmaroli**
- `Site#site_data` is called [*twice*](https://github.com/jekyll/jekyll-seo-tag/blob/f19b29ea0a20c08b32e3018800da83d9da69ace5/lib/jekyll-seo-tag/author_drop.rb#L63-L64) per page / document when the popular plugin `jekyll-seo-tag` is used
- `jekyll-seo-tag` is loaded by default via recent releases of Minima
- IMO, its better addressing this here instead of patching per plugin(s)

### CODE REVIEW

File: lib/jekyll/site.rb [hunk 1]
```diff
@@ -253,7 +253,7 @@ def categories
     #
     # Returns the Hash to be hooked to site.data.
     def site_data
-      config["data"] || data
+      @site_data ||= config["data"] || data
```
Comment (parkr) [inline lib/jekyll/site.rb]: If you want to memoize the `||`, then it'd be: `@site_data ||= (config["data"] || data)`. By default, this is equivalent to `(@site_data ||= config["data"]) || data`.

Comment (parkr) [reply lib/jekyll/site.rb]: This `@site_data` should also be reset in `reset`!

Comment (ashmaroli) [reply lib/jekyll/site.rb]: Regarding the use of parenthesis, I did not know, this would be interpreted that way.. Thanks!

Comment (parkr) [reply lib/jekyll/site.rb]: @ashmaroli You might enjoy http://tmm1.net/ruby21-profiling/ which has an example of where parentheses went awry!

Comment (ashmaroli) [reply lib/jekyll/site.rb]: Thanks. I'll check it out :smiley:

**Comment by oe** [comment]
@jekyllbot: merge +dev

**Pull request merged**